### PR TITLE
Run e2e tests on label change and introduce auto cancel of previous job

### LINKED ISSUE #558: Integrate turtles with `renovate-rancher` bot
### What would you like to be added (User Story)?

As a developer I’d like to ensure regular and automated dependency updates for the turtles code to fix security vulnerabilities in a timely manner with ability to adapt the workflow steps to match repository code and CI requirements.

### Detailed Description

Currently we are using dependabot workflow to make dependency bumps. While this approach is sufficient for the simple dependency updates, it is problematic in `go.mod` dependency bumps scenarios when there are multiple packages co-dependent on each other.

This way a dependency bump in one package requires a cascade update of the dependency in other packages. It is simple to achieve with a `make generate` task, but there is no place in a regular dependabot workflow to define user-specified steps to extend default handling.

We need to explore alternatives to extend regular uscase, using https://github.com/renovatebot as it is approved by [security](https://github.com/rancher/security-team/blob/main/docs/rancher-gha-standards.md#allowed-github-actions) or using [updatecli](https://github.com/updatecli).

An example configuration for rancher dependabot, as well as updatecli can be found in fleet:
- https://github.com/rancher/fleet/blob/main/.github/renovate.json
- https://github.com/rancher/fleet/blob/main/.github/workflows/renovate.yml
- https://github.com/rancher/fleet/tree/main/updatecli/updatecli.d
- https://github.com/rancher/fleet/blob/main/.github/workflows/updatecli.yml

### Anything else you would like to add?

_No response_

### Label(s) to be applied

/kind feature
/kind cleanup

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
**What this PR does / why we need it**:

- Ensure that e2e tests run on dependabot or update CLI pr, when the pr is labeled.
- Perform automatic cancel of previous running jobs, preventing long delays between job completion. 
- Label updatecli PRs with appropriate labels, which additionally triggers actions.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #558

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: updatecli/updatecli.d/manifest.yaml
Comment (salasberryfin) [inline updatecli/updatecli.d/manifest.yaml:166]:
```suggestion
        - area/update
```
I see we have an `area/update` label https://github.com/rancher/turtles/labels/area%2Fupdate that we can probably use here.

**Review by salasberryfin — COMMENTED** [review]
Thanks. Just a small comment on using existing labels for updatecli PRs.

File: updatecli/updatecli.d/manifest.yaml
Comment (Danil-Grigorev) [reply updatecli/updatecli.d/manifest.yaml]: Updated, thanks

**Review by salasberryfin — APPROVED** [review]
Thanks @Danil-Grigorev, LGTM

File: updatecli/updatecli.d/manifest.yaml
Comment (furkatgofurov7) [inline updatecli/updatecli.d/manifest.yaml:163]: if I am understanding correctly, we should avoid auto-merge on updatecli PRs since there could be a situation where it might bump to the version of provider compatible with x version of CAPI, but we still use x-1 version of CAPI in turtles or vice versa.

Comment (Danil-Grigorev) [reply updatecli/updatecli.d/manifest.yaml:163]: We have a number of required approvals, this will not be merged automatically. If such change gets into repo - it is after series of reviews and approves.

**Review by furkatgofurov7 — APPROVED** [review]

**Pull request merged**
